### PR TITLE
largest-series-product: Update cases_test.go

### PR DESCRIPTION
**Pull request by sbromberger**
Correct error messages to conform to expected results.

### CODE REVIEW

**Comment by bitfield** [comment]
This looks very helpful, @sbromberger, thank you!

One useful tip for Exercism PRs is it's a good idea to put the exercise name in the PR title, along with a very brief description of what the change does. For example:

> largest-series-product: make error messages more accurate/informative

Since almost all PRs to this repo consist of "updating cases_test.go", a little more information is really helpful to the reviewer.

File: exercises/largest-series-product/cases_test.go [hunk 1]
```diff
@@ -20,10 +20,10 @@ var tests = []struct {
 	{"73167176531330624919225119674426574742355349194934", 6, 23520, true, ""},
 	{"0000", 2, 0, true, ""},
 	{"99099", 3, 0, true, ""},
-	{"123", 4, -1, false, "span must be smaller than string length"},
+	{"123", 4, -1, false, "string length must be greater than or equal to span"},
```
Comment (bitfield) [inline exercises/largest-series-product/cases_test.go]: This is correct, but it sounds sort of back to front. Would it be better to say:

> span must be smaller than or equal to string length

File: exercises/largest-series-product/cases_test.go [hunk 2]
```diff
@@ -20,10 +20,10 @@ var tests = []struct {
...
 	{"", 0, 1, true, ""},
 	{"123", 0, 1, true, ""},
-	{"", 1, -1, false, "span must be smaller than string length"},
+	{"", 1, -1, false, "string length must be greater than or equal to span"},
```
Comment (bitfield) [inline exercises/largest-series-product/cases_test.go]: Ditto.

File: exercises/largest-series-product/cases_test.go [hunk 3]
```diff
@@ -20,10 +20,10 @@ var tests = []struct {
...
 	{"1234a5", 2, -1, false, "digits input must only contain digits"},
-	{"12345", -1, -1, false, "span must be greater than zero"},
+	{"12345", -1, -1, false, "span must be greater than or equal to zero"},
```
Comment (bitfield) [inline exercises/largest-series-product/cases_test.go]: We could maybe phrase this more concisely as:

> span must not be negative

**Comment by sbromberger** [comment]
I'm ok with whatever bikeshedding needs to happen. The problem for me was that the error messages are incorrect as currently specified, and that led to some confusing test results.

File: exercises/largest-series-product/cases_test.go (hunk 1 above)
Comment (sbromberger) [reply exercises/largest-series-product/cases_test.go]: "smaller" is a bit of a problem, I think, as it is primarily used as a dimensionality measurement, not a magnitude measurement. I prefer "less than" or "greater than", and the order is interchangeable from my perspective. Thoughts?

File: exercises/largest-series-product/cases_test.go (hunk 3 above)
Comment (sbromberger) [reply exercises/largest-series-product/cases_test.go]: I like that more.

**Comment by leenipper** [comment]
Since the cases_test.go is a generated file based on the [problem specifications canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/largest-series-product/canonical-data.json), and the error message text actually originates in the .json, I prefer letting a PR against [problem-specifications](https://github.com/exercism/problem-specifications) make this change. That will help keep the generator maintenance sane, and let error text improvements be shared with other tracks.

**Pull request closed without merging**